### ISSUE TITLE
mAP in tensorboard are all 0 or close to 0, even though the detector is able to detect many objects well

### ISSUE THREAD

**Issue by Karthik-Suresh93** [tensorflow/models#4331]
Please go to Stack Overflow for help and support:

http://stackoverflow.com/questions/tagged/tensorflow

Also, please understand that many of the models included in this repository are experimental and research-style code. If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **What is the top-level directory of the model you are using**: object_detection
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**:source
- **TensorFlow version (use command below)**:1.8
- **Bazel version (if compiling from source)**:-
- **CUDA/cuDNN version**:CUDA 9/ cuDNN 7
- **GPU model and memory**: Tesla K-80, (number:2). Each has a memory of 12GB
- **Exact command to reproduce**:tensorboard --logdir=eval/ 

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
I am training a faster rcnn model on a custom dataset using the object detection API. The training seems to be going well with losses currently:

INFO:tensorflow:global step 4766: loss = 2.8091 (2.764 sec/step)
INFO:tensorflow:global step 4767: loss = 1.6403 (2.768 sec/step)
INFO:tensorflow:global step 4767: loss = 1.6403 (2.768 sec/step)

However, when I try to visualize the metrics using `tensorboard --logdir=eval/`on tensorboard, I get mAPs in all categories either 0 or very close to 0 as shown below:

![map_tensorboard_0](https://user-images.githubusercontent.com/25709940/40321957-61ffc76c-5cf6-11e8-931a-217900ae9b3e.png)

When I look at the images tab in tensorboard, the detector seems to be finding decent boxes

![good_detection_4k](https://user-images.githubusercontent.com/25709940/40321998-88cdef9a-5cf6-11e8-8d70-bf7ecd80df84.png)

![night_time_4k](https://user-images.githubusercontent.com/25709940/40322005-9197ae54-5cf6-11e8-9691-70e932773ab1.png)

My configuration file is:

model {
  faster_rcnn {
    num_classes: 12
    image_resizer {
      keep_aspect_ratio_resizer {
        # Raw KITTI images have a resolution of 1242x375, if we wish to resize
        # them to have a height of 600 then their width should be
        # 1242/(375/600)=1987.2
        min_dimension: 1000
        max_dimension: 1600
      }
    }
    feature_extractor {
      type: 'faster_rcnn_resnet101'
      first_stage_features_stride: 16
    }
    first_stage_anchor_generator {
      grid_anchor_generator {
        scales: [0.25, 0.5, 1.0, 2.0]
        aspect_ratios: [0.5, 1.0, 2.0]
        height_stride: 16
        width_stride: 16
      }
    }
    first_stage_box_predictor_conv_hyperparams {
      op: CONV
      regularizer {
        l2_regularizer {
          weight: 0.0
        }
      }
      initializer {
        truncated_normal_initializer {
          stddev: 0.01
        }
      }
    }
    first_stage_nms_score_threshold: 0.0
    first_stage_nms_iou_threshold: 0.7
    first_stage_max_proposals: 300
    first_stage_localization_loss_weight: 2.0
    first_stage_objectness_loss_weight: 1.0
    initial_crop_size: 14
    maxpool_kernel_size: 2
    maxpool_stride: 2
    second_stage_box_predictor {
      mask_rcnn_box_predictor {
        use_dropout: false
        dropout_keep_probability: 1.0
        fc_hyperparams {
          op: FC
          regularizer {
            l2_regularizer {
              weight: 0.0
            }
          }
          initializer {
            variance_scaling_initializer {
              factor: 1.0
              uniform: true
              mode: FAN_AVG
            }
          }
        }
      }
    }
    second_stage_post_processing {
      batch_non_max_suppression {
        score_threshold: 0.0
        iou_threshold: 0.6
        max_detections_per_class: 100
        max_total_detections: 300
      }
      score_converter: SOFTMAX
    }
    second_stage_localization_loss_weight: 2.0
    second_stage_classification_loss_weight: 1.0
  }
}

train_config: {
  batch_size: 4
  optimizer {
    momentum_optimizer: {
      learning_rate: {
        manual_step_learning_rate {
          initial_learning_rate: 0.0001
          schedule {
            step: 500000
            learning_rate: .00001
          }
          schedule {
            step: 700000
            learning_rate: .000001
          }
        }
      }
      momentum_optimizer_value: 0.9
    }
    use_moving_average: false
  }
  gradient_clipping_by_norm: 10.0
  fine_tune_checkpoint: "faster_rcnn_resnet101_kitti_2018_01_28/model.ckpt"
  from_detection_checkpoint: true
  num_steps: 800000
  data_augmentation_options {
    random_horizontal_flip {
    }
  }
}

train_input_reader: {
  label_map_path:"data/object-detection.pbtxt"
  tf_record_input_reader: {
    input_path: "data/train_latest.record"
  }
}

eval_config: {
  use_moving_averages: false
  num_examples: 500
}

eval_input_reader: {
  label_map_path: "data/object-detection.pbtxt"
  tf_record_input_reader: {
    input_path: "data/val_latest.record"
  }
  }

Why are mAP values so bad? Please let me know if there is any other information I can provide.

Thanks and Regards,
Karthik

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by Karthik-Suresh93** [comment]
Can someone please help me with this issue?

**Comment by wywywy01** [comment]
i met this problem also.

**Comment by Karthik-Suresh93** [comment]
@wywywy01 

Were you able to solve it?

**Comment by wywywy01** [comment]
sorry ,i can't

**Comment by Karthik-Suresh93** [comment]
Which dataset are you training on? Does it have very small objects in it?

**Comment by wywywy01** [comment]
yes, it is, i modified some settings such as small scales and proper aspect_ratios, but the map is also 0.

except that settings, should i modify other settings too?

**Comment by Karthik-Suresh93** [comment]
I have read on other repos that removing very small objects (length <threshold, width<threshold) will help. I have not tried it yet

**Comment by Karthik-Suresh93** [comment]
This seems to be a bug definitely. I ran the same dataset with another repo and it works just fine.

**Comment by AakashKumarNain** [comment]
What is the smallest size of the objects that is allowed/counted for mAP calculation? Also, can you please point me to this page where this thing about small size objects is mentioned clearly?

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.

**Comment by canseloguzz** [comment]
can anyone solve this problem? ı have same too!
